src/main/java/zd/zero/waifu/motivator/plugin/listeners/Debouncer.kt

```diff
@@ -0,0 +1,22 @@
+package zd.zero.waifu.motivator.plugin.listeners
+
+import com.intellij.util.Alarm
+
+interface Debouncer {
+    fun debounce(toDebounce: Runnable)
+}
+
+class JetbrainsDebouncer(private val debounceInterval: Int) : Debouncer {
```

Comment:
minor: I'm thinking if we can rename this to a different one, as we're already in the Jetbrains context. What do you think if we rename this to `"NotificationDebouncer"` instead? This shows that the impl of `Debouncer` is used in the Notification context, and in case we increase the usage of this impl to multiple contexts we can rename it at that time.

Reply:
I can rename it. 

However, I've been using interfaces as a way to separate the implementation of the interface from the actual definition. That way I could implement a debouncer using an Alarm, a Timeout, or some other thing that would work.

The fact that I used it for notifications should not be reflected in the name of this class itself. We could potentially debounce other functions in the app, not just the notifications. So it feels weird coupling the name of the class to it's current use case. It's kind of like calling a hammer a Brass Nail Hammer because I was hammering brass nails at the time :)

I just called it JetbrainsDebouncer because I used a jetbrains platform API to implement it.

But that's just me.

Reply:
Sorry, I haven't made it clear. I was talking about renaming `JetbrainsDebouncer` to `NotificationDebouncer` and not the `Debouncer`.

> I just called it JetbrainsDebouncer because I used a jetbrains platform API to implement it.

Yes, this could be another argument I can accept ;)

> We could potentially debounce other functions in the app, not just the notifications. So it feels weird coupling the name of the class to it's current use case. It's kind of like calling a hammer a Brass Nail Hammer because I was hammering brass nails at the time :)

I agree with this, maybe the "Notification" was too specific, and I should've asked for something more generic like `EventDebouncer`.

I see that it was now renamed to `AlarmDebouncer`, we could definitely preserve that.